web/impact/impact/tests/test_create_edit_office_hours_view.py

```diff
@@ -0,0 +1,250 @@
+        self.assertEqual(response.data['detail'],
+                         DEFAULT_PERMISSION_DENIED_DETAIL)
+
+    def test_office_hour_start_date_must_be_later_than_start_date(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        start_time = datetime.now()
+        data = self._get_post_request_data(mentor, get_data={
+            'start_date_time': start_time,
+            'end_date_time': start_time + timedelta(minutes=-30),
+        })
+        response = self._create_office_hour_session(mentor, data)
+        self._assert_error_response(response.data,
+                                    key='end_date_time',
+                                    expected=INVALID_END_DATE)
+
+    def test_none_staff_or_none_mentor_response(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor)
+        response = self._create_office_hour_session(self.basic_user(), data)
+        self.assertEqual(response.data['detail'],
+                         DEFAULT_PERMISSION_DENIED_DETAIL)
+
+    def test_use_request_user_for_mentor_users(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        mentor2 = self._expert_user(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor,
+                                           get_data={'mentor': mentor2.id})
+        response = self._create_office_hour_session(mentor, data)
+        self.assertEqual(response.data['data']['mentor']['id'], mentor.id)
+
+    def test_admin_cant_create_office_hour_for_restricted_user_response(self):
+        user = self._expert_user(UserRole.JUDGE)
+        data = self._get_post_request_data(user)
+        response = self._create_office_hour_session(self.staff_user(), data)
+        self._assert_error_response(response.data,
+                                    key='mentor', expected=INVALID_USER)
+
+    def test_admin_cant_create_office_hour_for_restricted_user(self):
+        user = self._expert_user(UserRole.JUDGE)
+        data = self._get_post_request_data(user)
+        self._create_office_hour_session(self.staff_user(), data)
+        self._assert_office_hour_was_not_created(user)
+
+    def _expert_with_inactive_program(self, role):
+        program = ProgramFactory(program_status='ended')
+        return self._expert_user(role, program)
+
+    def _get_post_request_data(self, mentor, from_now=0, get_data=None):
+        location = LocationFactory()
+        start_time = datetime.now() + timedelta(minutes=from_now)
+        data = {
+            'mentor': mentor.id,
+            'start_date_time': start_time,
+            'end_date_time': start_time + timedelta(minutes=30),
+            'topics': 'topics',
+            'description': 'description',
+            'location': location.id
+        }
+        if get_data:
+            data.update(get_data)
+        return data
+
+    def _get_patch_request_data(self, office_hour, patch_data=None):
```

Comment:
Shouldn't a patch request be fine with just taking the changed fields?

Reply:
yeah, updating that.